parlai/core/worlds.py

```diff
@@ -907,13 +907,15 @@ def _get_task_world(opt):
             world_class = getattr(my_module, world_name)
         except Exception:
             # Defaults to this if you did not specify a world for your task.
-            world_class = DialogPartnerWorld
-    task_agents = _create_task_agents(opt)
+            if len(task_agents + user_agents) == 2:
+                world_class = DialogPartnerWorld
+            else:
+                world_class = MultiAgentDialogWorld
```

Comment:
nice--while we're here, we should also do this when `'.' in sp[0]` (line 896).

Reply:
@alexholdenmiller Done!